File: AUXILIATURA/INTRODUCCION A POO/Ejercicio3/codigo/ejercicio3.py
#3.Crea una clase Coche con marca, modelo y velocidad
#    a) Agrega un método acelerar () que aumente la velocidad en 10
#    b) Agrega un método frenar () que disminuya la velocidad en 5
#    c) Crea dos coches, aceléralos, frénalos y muestra sus velocidadesclass Coche:
    def __init__(self, marca, modelo):
        self.marca = marca
        self.modelo = modelo
        self.velocidad = 0
    
    def acelerar(self):
        self.velocidad += 10
    
    def frenar(self):
        self.velocidad = max(0, self.velocidad - 5)
    
    def mostrar_velocidad(self):
        print(f"{self.marca} {self.modelo}: Velocidad actual = {self.velocidad} km/h")

coche1 = Coche("Toyota", "Corolla")
coche2 = Coche("Ford", "Mustang")

coche1.acelerar()
coche2.acelerar()
coche2.acelerar() 

coche1.frenar()
coche2.frenar()

coche1.mostrar_velocidad()
coche2.mostrar_velocidad()

File: AUXILIATURA/INTRODUCCION A POO/Ejercicio3/codigo/ejercicio3.java

public class Coche {
    private String marca;
    private String modelo;
    private int velocidad;
    public Coche(String marca, String modelo) {
        this.marca = marca;
        this.modelo = modelo;
        this.velocidad = 0;
    }
    public void acelerar() {
        velocidad += 10;
    }
    public void frenar() {
        velocidad = Math.max(0, velocidad - 5);
    }
    public void mostrarVelocidad() {
        System.out.printf("%s %s: Velocidad actual = %d km/h%n", marca, modelo, velocidad);
    }
    public static void main(String[] args) {
        Coche coche1 = new Coche("Toyota", "Corolla");
        Coche coche2 = new Coche("Ford", "Mustang");
        
      
        coche1.acelerar();
        coche2.acelerar();
        coche2.acelerar(); 
        
        coche1.frenar();
        coche2.frenar();
        
        coche1.mostrarVelocidad();
        coche2.mostrarVelocidad();
    }
}
